suricata/update/rule.py

```diff
@@ -146,6 +146,8 @@ def __str__(self):
         return self.format()
 
     def format(self):
+        if self.noalert and not "noalert;" in self.raw:
+            self.raw = re.sub(r'( *sid\: *[0-9]+\;)', r' noalert;\1', self.raw)
```

Comment:
does this mean a space between 'sid:' and the value is required?

Reply:
No, it's not required. It's just optional so that the regex doesn't fail if someone wrote a rule with a space between "sid:" and the number, eg. "sid: 1234567;".

Reply:
Ok thanks. @shivan1b do we have a way to add an automated test?

Reply:
Yes. We can add a test method to `test_rule.py` satisfying the conditions above in the `format()` method. We could also modify existing test `test_toggle_rule` with a new usecase for `noalert` set to True.
@King-Konsto do you mind adding a test? Let me know if you need any assistance with that.

Reply:
Yes, I can do that. Can you guide me to an example of an existing test and explain what you think I should do please?

Reply:
One template test could be `def test_disable_rule(self)`.  Parse a rule, set the noalert on it, reformat it, and make sure it parses again?

Reply:
Also, @King-Konsto it would be great if you could add a test for `resolve_flowbits` function since that is the function where you actually made a change. In order to write the test, try to replicate the function signature (here `rulemap` and `disabled_rules`). These should be hardcoded by you so that tests do not give different results in different settings.
You can take reference from any test https://github.com/OISF/suricata-update/blob/master/tests/test_rule.py to understand what your test should look like. Let me know if you need any more assistance.

Reply:
Thanks for your help. I have added test cases. Please have a look at https://github.com/OISF/suricata-update/pull/147